modules/core/shared/src/main/scala/gem/Target.scala

```diff
@@ -3,5 +3,23 @@
 
 package gem
 
+import cats.Eq
+
+import gem.enum.Site
+import gem.math.Ephemeris
+
+import monocle.Optional
+import monocle.macros.Lenses
+
 /** A target of observation. */
-final case class Target(name: String, track: Track)
+@Lenses final case class Target(name: String, track: Track)
+
+@SuppressWarnings(Array("org.wartremover.warts.PublicInference"))
+object Target {
+
+  val ephemerides: Optional[Target, Map[Site, Ephemeris]] =
+    track composeOptional Track.ephemerides
+
+  implicit val EqTarget: Eq[Target] =
```

Comment:
If we can define `Order[Target]` we can use `TreeSet` in the `TargetEnvironment` which is a bit more sane. It would also give us a deterministic ordering which might be good. We also may be able to define that traversal you were looking for. Checking with Julien.

Reply:
I'll circle back to the `Order[Target]` and `TreeSet` idea.